packages/contracts/contracts/DependencyRegistryV0.sol

```diff
@@ -1023,16 +1001,13 @@ contract DependencyRegistryV0 is
         view
         returns (address[] memory)
     {
-        DependencyRegistryStorageLib.Storage
-            storage ds = DependencyRegistryStorageLib.s();
-
-        uint256 supportedCoreContractCount = ds.supportedCoreContracts.length();
-        address[] memory supportedCoreContracts = new address[](
-            supportedCoreContractCount
-        );
+        uint256 count = ICoreRegistryV1(coreRegistryAddress)
+            .getNumRegisteredContracts();
+        address[] memory supportedCoreContracts = new address[](count);
 
-        for (uint256 i; i < supportedCoreContractCount; ) {
-            supportedCoreContracts[i] = ds.supportedCoreContracts.at(i);
+        for (uint256 i; i < count; ) {
+            supportedCoreContracts[i] = ICoreRegistryV1(coreRegistryAddress)
+                .getRegisteredContractAt(i);
```

Comment:
can use the core registry function `getAllRegisteredContracts`

Reply:
ah perfect, i updated this (and added getAllRegisteredContracts to the core registry interface)